app/javascript/tests/public_health/bulk_actions/UpdateAssignedUser.test.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import InfoTooltip from '../../../components/util/InfoTooltip';
+import UpdateAssignedUser from '../../../components/public_health/bulk_actions/UpdateAssignedUser';
+import { mockPatient1, mockPatient2, mockPatient6 } from '../../mocks/mockPatients';
+
+const onCloseMock = jest.fn();
+const mockToken = 'testMockTokenString12345';
+const mockAssignedUserDatalist = ['3682', '903874', '73892', '1689'];
+
+function getWrapper(patientsArray) {
+  let wrapper = shallow(<UpdateAssignedUser authenticity_token={mockToken} patients={patientsArray} close={onCloseMock} assigned_users={mockAssignedUserDatalist} />);
+  return wrapper;
+}
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('UpdateAssignedUser', () => {
+  it('Sets intial state after mount correctly', () => {
+    const wrapper = getWrapper([mockPatient1, mockPatient2]);
+    // componentDidMount is called when mounted so a timeout is needed to give it time to resolve.
+    setTimeout(() => {
+      expect(wrapper.state('assigned_user')).toEqual('21');
+      expect(wrapper.state('initial_assigned_user')).toEqual('21');
+      expect(wrapper.state('apply_to_household')).toBeFalsy();
+      expect(wrapper.state('loading')).toBeFalsy();
+    }, 100);
+  });
+
+  it('Properly renders all main components', () => {
+    const wrapper = getWrapper([mockPatient1, mockPatient2]);
+    expect(wrapper.find('ModalBody').find('div').at(0).text()).toContain('Please input the desired Assigned User to be associated with all selected monitorees:');
+    expect(wrapper.find(InfoTooltip).exists()).toBeTruthy();
+    expect(wrapper.find('#assigned_user_input').exists()).toBeTruthy();
+    expect(wrapper.find('#assigned_users').exists()).toBeTruthy();
+    wrapper
+      .find('#assigned_users')
+      .find('option')
+      .forEach((option, index) => {
+        expect(option.text()).toEqual(mockAssignedUserDatalist[Number(index)]);
+      });
+    expect(wrapper.find('#apply_to_household').exists()).toBeTruthy();
+    expect(wrapper.find('ModalFooter').find('Button').at(0).text()).toEqual('Cancel');
+    expect(wrapper.find('ModalFooter').find('Button').at(1).text()).toEqual('Submit');
+    // componentDidMount is called when mounted so a timeout is needed to give it time to resolve.
+    setTimeout(() => {
```

Comment:
same comment here about the set timeout, if there is a way to avoid it throughout this file that would be ideal